### DISCUSSION TITLE
Nullable Reference Types and Should().NotBeNull()

### DISCUSSION THREAD

**Discussion by mu88** [fluentassertions/fluentassertions#1750] [Q&A] [answered]
Hey guys 👋🏻

When using Nullable Reference Types, the following code gives me a compiler warning `Dereference of a possibly null reference` on `testee.StatusCode`:
```cs
[Fact]
public void Test()
{
    var testee = GetHttpResponse();

    testee.Should().NotBeNull();
    testee.StatusCode.Should().Be(HttpStatusCode.Accepted);
}

private HttpResponseMessage? GetHttpResponse() => new HttpResponseMessage(HttpStatusCode.Accepted);
```

Since I'm doing an explicit null check before, this warning is a false positive.

Can the [null-state analysis attributes](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis) help us to avoid the misleading warning?

**Comment by dennisdoomen** [comment, marked as answer]
It's a difficult discussion. See https://github.com/fluentassertions/fluentassertions/issues/1115

**Reply by mu88** [reply]
Oh I see, thanks Dennis
